src/main/resources/hudson/tasks/junit/History/index.jelly

```diff
@@ -24,84 +24,127 @@ THE SOFTWARE.
-        <st:bind value="${it}"/>
-      </script>
-      <script type="text/javascript" src="${resURL}/plugin/junit/history/history.js"/>
     </st:once>
 
     <st:include from="${it.testObject}" it="${it.testObject}" page="sidepanel.jelly"/>
     <l:main-panel>
-      <j:choose>
-        <j:when test="${it.historyAvailable()}">
-          <bs:card title="${%History}" fontAwesomeIcon="chart-line">
-            <div id="trend-carousel" class="carousel slide carousel-dark" data-bs-interval="false">
-              <div class="carousel-inner">
-                <div class="carousel-item">
-                  <div id="test-duration-trend-chart" class="graph-cursor-pointer card-chart-carousel"/>
-                </div>
-                <div class="carousel-item active">
-                  <div id="test-result-trend-chart" class="graph-cursor-pointer card-chart-carousel"/>
-                </div>
-              </div>
-              <button class="carousel-control-prev" type="button" data-bs-target="#trend-carousel" data-bs-slide="prev">
-                <span class="carousel-control-prev-icon" aria-hidden="true"/>
-                <span class="visually-hidden">Previous</span>
-              </button>
-              <button class="carousel-control-next" type="button" data-bs-target="#trend-carousel" data-bs-slide="next">
-                <span class="carousel-control-next-icon" aria-hidden="true"/>
-                <span class="visually-hidden">Next</span>
-              </button>
-            </div>
-          </bs:card>
+      <div
+        id="history-data"
+        data-start="${start}"
+        data-end="${end}"
+        data-count="${count}"
+        data-trendChartJson="${historySummary.trendChartJson}"
+        data-appRootUrl="${app.rootUrl}"
+        data-testObjectUrl="${it.testObject.url}"
+        class="jenkins-!-margin-bottom-2 jenkins-!-margin-top-0"
+        style="display: flex; flex-direction: row; justify-content: space-between; align-items: center; height: auto;"
+        >
+        <j:if test="${it.testObject.run.parent.builds.size() > end}">
+          <a href="${app.rootUrl}${it.testObject.url}/history?start=${end+1}&amp;count=${count}"
+              class="jenkins-!-margin-right-2" style="height: fit-content">${%Older}
+          </a>
+        </j:if>
 
-        </j:when>
-        <j:otherwise>
-          ${%More than 1 builds are needed for the chart.}
-        </j:otherwise>
-      </j:choose>
+        <j:if test="${start > 0}">
+          <a href="${app.rootUrl}${it.testObject.url}/history${(start-count)>0?'?start='+(start-count)+'&amp;':'?'}count=${count}"
+              class="jenkins-!-margin-right-2" style="height: fit-content">${%Newer}
+          </a>
+        </j:if>
+        <div style="flex-grow: 1"/>
+        <div style="display: flex; flex-basis: auto; justify-content: space-between; align-items: center" tooltip="The history window size. Out of N builds, only the builds with test results are shown.">
```

Comment:
somewhere around here:

<img width="335" alt="image" src="https://github.com/user-attachments/assets/c747ec93-0002-48ec-9d48-811c319339b1">

Icons and legend are overlapping each other when viewing on a mobile device. Would be ideal to fix this

Reply:
I think Trend and Smooth graphs need to go away on narrow screens too, not sure if I can fix the button layout of ECharts, it seems to come from them.

Reply:
Interestingly, it works way better if I do this

```
    <st:once>
      <c:chart-setup id="test-history"/>
      <script id="history-data-2" data-trendChartJson="${historySummary.trendChartJson}"></script>
    </st:once>
```

This seems to feel like before, but it is now a dummy empty script tag. Is this acceptable?